### PR TITLE
Bug 1809921 - Address crash when visiting internal sites like about config when cookie banner is ON

### PR DESCRIPTION
**Pull request by Amejia481**
To reproduce visit, `about:config` when cookie banner handling is ON, then tab the shield icon.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

**Review by iorgamgabriel — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/settings/quicksettings/protections/cookiebanners/CookieBannerUtil.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * Queries [CookieBannersStorage.hasException] and handles unexpected exceptions.
+ */
+@Suppress("TooGenericExceptionCaught")
+suspend inline fun CookieBannersStorage.safeHasException(tab: SessionState, logger: Logger): Boolean {
+    return try {
+        hasException(
+            tab.content.url,
+            tab.content.private,
+        )
+    } catch (e: Exception) {
```
Comment (iorgamgabriel) [inline app/src/main/java/org/mozilla/fenix/settings/quicksettings/protections/cookiebanners/CookieBannerUtil.kt:21]: I think it's a good idea to throw the message of the Exception . Maybe a different exception happens here .

**Comment by iorgamgabriel** [comment]
This crash happens both in Fenix and Focus. Maybe it will be a good idea to upstream the method safeHasException in Android Components because now I have to do the same fix in Focus.

File: app/src/main/java/org/mozilla/fenix/settings/quicksettings/protections/cookiebanners/CookieBannerUtil.kt (hunk 1 above)
Comment (Amejia481) [reply app/src/main/java/org/mozilla/fenix/settings/quicksettings/protections/cookiebanners/CookieBannerUtil.kt:21]: Thanks I'll add the exception as part of the log!

**Comment by Amejia481** [comment]
> This crash happens both in Fenix and Focus. Maybe it will be a good idea to upstream the method safeHasException in Android Components because now I have to do the same fix in Focus. @Amejia481

We normally don't handle exceptions in AC as apps could handle the exceptions differently.

**Comment by Amejia481** [comment]
I'll move it to ac 👍🏽

**Comment by iorgamgabriel** [comment]
Thank you @Amejia481

**Comment by Amejia481** [comment]
Close in favour of https://github.com/mozilla-mobile/firefox-android/pull/554

**Pull request closed without merging**